Tests/WitsmlExplorer.Api.Tests/Services/CredentialsCacheTests.cs

```diff
@@ -20,42 +21,97 @@ public CredentialsCacheTests()
             _credentialsCache = new CredentialsCache(logger.Object);
         }
 
-
         [Fact]
         public void RemoveAllClientCredentials_TwoOfThree_ReturnOne()
         {
-            string clientId = Guid.NewGuid().ToString();
+            string cacheId = Guid.NewGuid().ToString();
             string url = "https://somehost";
             string witsmlUsername = "user";
 
             _credentialsCache.Clear();
-            _credentialsCache.SetItem($"{clientId}@{url}{_random.Next(1000)}.com", $"DUMMY_VALUE{_random.Next(1000)}", 1.0, witsmlUsername);
-            _credentialsCache.SetItem($"{clientId}@{url}{_random.Next(1000)}.com", $"DUMMY_VALUE{_random.Next(1000)}", 1.0, witsmlUsername);
-            _credentialsCache.SetItem($"{Guid.NewGuid()}@{url}{_random.Next(1000)}.com", $"DUMMY_VALUE{_random.Next(1000)}", 1.0, witsmlUsername);
+            _credentialsCache.SetItem(cacheId, new Uri($"{url}{_random.Next(1000)}.com"), $"DUMMY_VALUE{_random.Next(1000)}", 1.0, witsmlUsername);
+            _credentialsCache.SetItem(cacheId, new Uri($"{url}{_random.Next(1000)}.com"), $"DUMMY_VALUE{_random.Next(1000)}", 1.0, witsmlUsername);
+            _credentialsCache.SetItem(Guid.NewGuid().ToString(), new Uri($"{url}{_random.Next(1000)}.com"), $"DUMMY_VALUE{_random.Next(1000)}", 1.0, witsmlUsername);
 
             long before = _credentialsCache.Count();
-            _credentialsCache.RemoveAllClientCredentials(clientId);
+            _credentialsCache.RemoveAllClientCredentials(cacheId);
             long after = _credentialsCache.Count();
-            Assert.Equal(3, before);
+            Assert.Equal(2, before);
             Assert.Equal(1, after);
             _credentialsCache.Clear();
         }
+
         [Fact]
         public void Clear_SetThree_ReturnZero()
```

Comment:
Maybe the signature should be changed a bit because we set two duplicates and one unique cache item resulting in just setting two. and asserting two.
Maybe either named `Clear_SetTwo_ReturnZero` or `Clear_SetThreeWithTeoDuplicates_ReturnZero` ? What do you think?

Reply:
I don't think the input matters much when we test the clear method. I can adjust it.